src/modules/types.rs

```diff
@@ -107,3 +130,61 @@ pub fn try_parse_type(meta: &mut ParserMetadata) -> Result<Type, Failure> {
 
     res
 }
+
+#[cfg(test)]
+mod tests {
```

Comment:
is there a reason to put tests in this file? i think they should be in the `src/tests` directory, along with all the other ones

Reply:
Isn't it idiomatic to put unit tests in the same files?

Reply:
Yes, it is. The tests directory is for integration tests, and these are simply unit tests.